File: python/BillsApi/app/application/bills/ports/bill_write_repository.py
from abc import ABC, abstractmethod

from app.domain.bills.entities import NewBill


class BillWriteRepository(ABC):
    @abstractmethod
    def exists_by_bill_number(self, bill_number: str) -> bool:
        raise NotImplementedError

    @abstractmethod
    def create(self, new_bill: NewBill) -> int:
        raise NotImplementedError

File: node-ts/BillsApi/src/application/bills/create/BillWriteRepository.ts
import { NewBill } from "../../../domain/bills/NewBill";
import { CreateBillResult } from "./CreateBillResult";

export interface BillWriteRepository {
  existsByBillNumber(billNumber: string): Promise<boolean>;
  createInTransaction(newBill: NewBill): Promise<CreateBillResult>;
}
